### PR TITLE
Converted the classic pipeline to yaml Azure Templates CI

### PR DESCRIPTION
**Pull request by Harinder-accolite**
classic pipeline:-

repository: https://github.com/lansa/azure-quickstart-templates
branch: support/scalable
pipeline name - Template CI

converted classic to yaml pipeline

repository: Harinder-accolite/azure-quickstart-templates
branch: feature/harinder-classic2yaml
pipeline name: Azure Templates CI 
pipeline url: https://dev.azure.com/VisualLansa/Lansa%20Azure%20Scalable%20License%20Images/_build/results?buildId=2116&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a

### CODE REVIEW

**Comment by Harinder-accolite** [comment]
Created the task to delete resource group as the Sql  Server (build-temsqlserver) was not getting deleted during the pipeline run.Is it okay that resource group is delete or we need to modify and delete only Sql Server.

**Comment by robe070** [comment]
> Created the task to delete resource group as the Sql Server (build-temsqlserver) was not getting deleted during the pipeline run.Is it okay that resource group is delete or we need to modify and delete only Sql Server.

Yes this is exactly what is required. Remove EVERYTHING that the creation of the Resource Group created. This is presuming that the resource group name is unique to this pipeline.

**Comment by Harinder-accolite** [comment]
We have added the task to delete the resource group only when all the required task are successful.

**Review by robe070 — CHANGES_REQUESTED** [review]
As you may not delete it, you also need to delete it if it exists BEFORE you deploy the template. This means that you cannot include the Build Number in the resource group name.

Change the name the Resource Group to 
```
BakingDP-Templates-CI-Test
```

And add a Task to delete the RG BEFORE deploying the template - and if it does not exist that is FINE. Ignore the error.

**Comment by Harinder-accolite** [comment]
Moved the Task for task for RG deletion and replaced the RG name to "BakingDP-Templates-CI-Test" also added the condition to continue if no RG is available if a resource is available the task will delete the RG and then Deploy the Template.

File: Azure Template CI.yaml [hunk 1]
```diff
@@ -328,10 +338,3 @@ jobs:
       ArtifactName: Solution Template
       FileCopyOptions: ''
 
-  - task: AzureResourceManagerTemplateDeployment@3
-    condition: succeeded()
-    inputs:
-      deploymentScope: 'Resource Group'
-      ConnectedServiceName: $(ConnectedServiceName)
-      action: 'DeleteRG'
-      resourceGroupName: BakingDP-TemplateTest-$(Build.BuildNumber)
```
Comment (robe070) [inline Azure Template CI.yaml]: I said to duplicate the task, not move it. You need to re-instate this task as it is.

File: Azure Template CI.yaml [hunk 2]
```diff
@@ -77,13 +77,23 @@ jobs:
...
       pwsh: true
 
+  - task: AzureResourceManagerTemplateDeployment@3
+    displayName: Delete resource group BakingDP-Templates-CI-Test
+    condition: succeeded()
+    inputs:
+      deploymentScope: 'Resource Group'
+      ConnectedServiceName: $(ConnectedServiceName)
+      action: 'DeleteRG'
+      resourceGroupName: BakingDP-Templates-CI-Test
```
Comment (robe070) [inline Azure Template CI.yaml]: You need to make the RG Name a YAML variable.

**Review by robe070 — CHANGES_REQUESTED** [review]
Hopefully you understand my comments this time.

File: Azure Template CI.yaml (hunk 2 above)
Comment (Harinder-accolite) [reply Azure Template CI.yaml]: Created the RG Name as YAML variable.

File: Azure Template CI.yaml (hunk 1 above)
Comment (robe070) [reply Azure Template CI.yaml]: You have still not done this!

**Review by robe070 — CHANGES_REQUESTED** [review]
The changes you made are fine, you've left one out, as I've separately commented.

**Comment by Harinder-accolite** [comment]
Changes were made i.e task has been duplicated on the yaml and pushed to pipeline.

**Comment by Harinder-accolite** [comment]
As per the discussion on the Call have deleted the task to continue on error at end of pipeline where as at the starting this task works fine.Ran the pipeline and artifact were created successfully .

**Comment by robe070** [comment]
Needed to change literal reference to resource group instead of $(ResourceGroup)

**Comment by nkhl-accolite** [comment]
@robe070 - Do you mean you want to replace every "$(ResourceGroup)" with the actual value, which is "_BakingDP-Templates-CI-Test_"

**Comment by robe070** [comment]
What needs fixing? How will you know you've fixed it?

Leave it as it is. Run the pipeline. What's the symptom?

**Pull request merged**